### DISCUSSION TITLE
Include header when copying

### DISCUSSION THREAD

**Discussion by nicosayer** [nick-keller/react-datasheet-grid#62] [Feature requests]
Love the repo, congrats on your work !

We use it a lot to copy and paste data from/to excel.

The only issue we have is when copying data from react-datasheet-grid to Excel, the header titles cannot be included in the clipboard and we have to manually add them in our Excel.

<img width="893" alt="Capture_d’écran_2021-10-04_à_15_33_25" src="https://user-images.githubusercontent.com/23241961/135861427-8df5f8cc-96a2-459c-ab91-3e2994abf19d.png">

Do you think it would be doable to add a `includeTitleWhenCopying` props on react-datasheet-grid ?

**Comment by nick-keller** [comment]
I understand the goal, but I do not believe this is a viable solution.
If the user copies the headers, then the user cannot paste the data onto the grid properly, the headers will end up in the grid (when the data is compatible). This defeats the purpose of DSG.

I would suggest a button outside of the grid (could even be in the footer) that copies the data with headers when you click it.
- You would have to construct the clipboard content yourself 
- You can use a ref to the grid to know which rows are selected (if any) and only copy the selected rows
- You could also add an item to the context menu to display "Copy with headers"

I hope that helps for now, if you can come up with another solution I'll gladly consider it !

**Reply by nicosayer** [reply]
I will add the button on my side. Thanks for the answer !

**Reply by grazielleanna** [reply]
Hi! Did you manage to make this button to insert the header when copying?

**Reply by nicosayer** [reply]
@grazielleanna, I finally dropped the idea so I haven't tried this.
